src/App.tsx

```diff
@@ -1,61 +1,164 @@
+  const [errorsInForm, setErrorsInForm] = useState(formFieldsErrors());
+  const [newTodoForm, setNewTodoForm] = useState({ ...formFields });
+  const [isTouched, setIsTouched] = useState(false);
+
+  const handleValidation = (newTodo: FormFields) => {
+    for (const key in newTodo) {
+      const isError = newTodo[key] === formFields[key];
+
+      if (isError) {
+        setErrorsInForm(prevState => {
+          return {
+            ...prevState,
+            [key]: true,
+          };
+        });
+      }
+    }
+  };
+
+  const handleErrorAppear = (isIncorrectField: boolean) => {
+    return isIncorrectField && isTouched;
+  };
+
+  const handleClearForm = () => {
+    setNewTodoForm({ ...formFields });
+  };
+
+  const handleFormChange = (fieldName: string, value: string) => {
+    setErrorsInForm(prevState => {
+      return {
+        ...prevState,
+        [fieldName]: false,
+      };
+    });
+    setNewTodoForm(prevState => {
+      return {
+        ...prevState,
+        [fieldName]: value,
+      };
+    });
+  };
+
+  const mistakesInForm =
+    newTodoForm.title === formFields.title ||
+    newTodoForm.user === formFields.user;
+
+  const handleAddTodo = (
+    event: React.FormEvent<HTMLFormElement>,
+    todo: FormFields,
+  ): void => {
+    event.preventDefault();
+    setIsTouched(true);
+    handleValidation(newTodoForm);
+
+    if (mistakesInForm) {
+      return;
+    }
+
+    const newTodo = makeNewTodo(
+      Math.max(...todoList.map(task => task.id)) + 1,
+      todo.title,
+      +todo.user,
+    );
+
+    setTodoList(prevState => {
+      return [...prevState, newTodo];
+    });
+
+    handleClearForm();
+    setIsTouched(false);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
+      <form onSubmit={event => handleAddTodo(event, newTodoForm)}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            <span>Title: </span>
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={newTodoForm.title}
+              onChange={({ target }) =>
+                handleFormChange('title', target.value.trimStart())
+              }
+            />
+          </label>
+          {handleErrorAppear(errorsInForm.title) && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
+          <label>
+            <span>User: </span>
+            <select
+              value={newTodoForm.user}
+              data-cy="userSelect"
+              onChange={event => handleFormChange('user', event.target.value)}
```

Comment:
[CODE STYLE] - Don't use one-letter variable naming. Instead of `event => handleFormChange('user', event.target.value)`, use a more descriptive variable name like `changeEvent`. This will make the code clearer to other developers.